ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -975,6 +977,138 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
+
+    def _on_span_started(span: Span):
+        inputs = get_argument_value(args, kwargs, 0, "input")
+        if not integration.is_pc_sampled_span(span):
+            return
+        if not isinstance(inputs, list):
+            inputs = [inputs]
+        for idx, inp in enumerate(inputs):
+            if not isinstance(inp, dict):
+                span.set_tag_str("langchain.request.inputs.%d" % idx, integration.trunc(str(inp)))
+                continue
+            for k, v in inp.items():
+                span.set_tag_str("langchain.request.inputs.%d.%s" % (idx, k), integration.trunc(str(v)))
+
+    def _on_span_finished(span: Span, streamed_chunks):
+        if span.error or not integration.is_pc_sampled_span(span):
+            return
+        if (
+            streamed_chunks
+            and langchain_core
+            and isinstance(instance.steps[-1], langchain_core.output_parsers.JsonOutputParser)
+        ):
+            # it's possible that the chain has a json output parser
+            # this will have already concatenated the chunks into a json object
+
+            # it's also possible the json output parser isn't the last step,
+            # but one of the last steps, in which case we won't act on it here
+            content = json.dumps(streamed_chunks[-1])
+        else:
+            # best effort to join chunks together
+            content = "".join([str(chunk) for chunk in streamed_chunks])
+        span.set_tag_str("langchain.response.content", integration.trunc(content))
+
+    return shared_stream(
+        integration=integration,
+        pin=pin,
+        func=func,
+        instance=instance,
+        args=args,
+        kwargs=kwargs,
+        interface_type="chain",
+        on_span_started=_on_span_started,
+        on_span_finished=_on_span_finished,
+    )
+
+
+@with_traced_module
+def traced_chat_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+    llm_provider = instance._llm_type
+
+    def _on_span_started(span: Span):
+        if not integration.is_pc_sampled_span(span):
+            return
+        chat_messages = get_argument_value(args, kwargs, 0, "input")
+        tag_general_message_input(span, chat_messages, integration, langchain_core)
+
+        for param, val in getattr(instance, "_identifying_params", {}).items():
+            if not isinstance(val, dict):
+                span.set_tag_str("langchain.request.%s.parameters.%s" % (llm_provider, param), str(val))
+                continue
+            for k, v in val.items():
+                span.set_tag_str("langchain.request.%s.parameters.%s.%s" % (llm_provider, param, k), str(v))
+
+    def _on_span_finished(span: Span, streamed_chunks):
+        if span.error or not integration.is_pc_sampled_span(span):
+            return
+        content = "".join([str(chunk.content) for chunk in streamed_chunks])
```

Comment:
Just to be safe, is `chunk` guaranteed to have the `content` attribute?

Reply:
yeah, `model.stream` for any generic `model` from a langchain partner library uses the base definition of `stream` in `langchain_core`, which returns an iterator of `BaseMessage` chunks, which have `content` on them. But I agree it's best to be safe, I'll use `getattr` with a default of the chunk itself and stringify

Reply:
added a safety check in https://github.com/DataDog/dd-trace-py/pull/10672/commits/47a3f15fdc43b840d42c6749b0a9ed2681c68a19